primitives/arithmetic/src/traits.rs

```diff
@@ -153,6 +153,12 @@ pub trait AtLeast32BitUnsigned: AtLeast32Bit + Unsigned {}
 
 impl<T: AtLeast32Bit + Unsigned> AtLeast32BitUnsigned for T {}
 
+// A meta trait for arithmetic. Only allows for unsigned integers, at most of 32 bits, the opposite of [`AtLeast32BitUnsigned`]
+pub trait AtMost32BitUnsigned: Into<u32> + Unsigned {}
+
+impl<T: Into<u32> + Unsigned> AtMost32BitUnsigned for T {}
```

Comment:
This has the same issue as you have explained elsewhere: A lot of things are Into<u32>, that are not numbers. Should only and manually be implemented for u8, u16 and u32.

Reply:
Even if you make `Into<u32> + Unsigned`? The `Unsigned` trait also impls `Num`, isn't this sufficient to guarantee that we are dealing with 'unsigned numbers'?